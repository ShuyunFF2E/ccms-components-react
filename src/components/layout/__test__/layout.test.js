import React from 'react';
import { mount, render } from 'enzyme';
import Layout from '../index';

const { Header, Content, Footer, Sider } = Layout;

describe('Layout', () => {

    it('basic layout should render correctly', () => {
        const wrapper = render((
            <Layout>
                <Header>header</Header>
                <Content>content</Content>
                <Footer>footer</Footer>
            </Layout>
        ));
        expect(wrapper).toMatchSnapshot();
    });
    it('left to right layout should render correctly', () => {
        const wrapper = render((
            <Layout hasSider>
                <Sider>Sider</Sider>
                <Layout>
                    <Header>Header</Header>
                    <Content>Content</Content>
                    <Footer>Footer</Footer>
                </Layout>
            </Layout>
        ));
        expect(wrapper).toMatchSnapshot();
    });
    it('layout property hasSider', () => {
        console.log(document);
        const wrapper = mount((
            <Layout  hasSider>
                <Sider>Sider</Sider>
                <Layout>
                    <Header>Header</Header>
                    <Content>Content</Content>
                    <Footer>Footer</Footer>
                </Layout>
            </Layout>
        ));
        const hasSide = wrapper.find('section.cloud-layout-has-sider');
        expect(hasSide).toBeTruthy();
    })

});
