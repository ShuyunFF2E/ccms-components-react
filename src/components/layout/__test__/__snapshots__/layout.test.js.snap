// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout basic layout should render correctly 1`] = `
<section
  class="cloud-layout"
>
  <header
    class="cloud-layout-header"
  >
    header
  </header>
  <main
    class="cloud-layout-content"
  >
    content
  </main>
  <footer
    class="cloud-layout-footer"
  >
    footer
  </footer>
</section>
`;

exports[`Layout left to right layout should render correctly 1`] = `
<section
  class="cloud-layout cloud-layout-has-sider"
>
  <aside
    class="cloud-layout-sider"
  >
    Sider
  </aside>
  <section
    class="cloud-layout"
  >
    <header
      class="cloud-layout-header"
    >
      Header
    </header>
    <main
      class="cloud-layout-content"
    >
      Content
    </main>
    <footer
      class="cloud-layout-footer"
    >
      Footer
    </footer>
  </section>
</section>
`;
